### ISSUE TITLE
Bugs in the examples folder

### ISSUE THREAD

**Issue by AlexandreAdam** [aymgal/COOLEST#35]
Hi, I am reviewing this software for JOSS. I am encountering difficulties when running the examples provided in the `/examples` folder.

## Relevant system info
Running on `python==3.8.5`

```
Package                            Version             Location
---------------------------------- ------------------- ----------------------------------------------
astropy                            5.2.1
coolest                            0.0.1               /home/alexandre/Desktop/Projects/COOLEST
numpy                              1.22.3
jsonpickle                         3.0.1
```

## `generate_all_profiles.py`
When running `python generate_all_profiles.py` from the examples directory, I get the following error
```
Traceback (most recent call last):
  File "generate_all_profiles.py", line 31, in <module>
    encoder_json = JSONSerializer('all_mass_profiles', obj=all_mass_profiles, indent=2)
  File "/home/alexandre/Desktop/Projects/COOLEST/coolest/template/json.py", line 50, in __init__
    raise ValueError("Path to JSON file must be an absolute path")
ValueError: Path to JSON file must be an absolute path
```
I get this behavior from the main branch and the JOSS branch. This [stackoverflow post](https://stackoverflow.com/questions/72289158/error-trial-writing-csv-is-not-absolute-all-paths-must-be-absolute) might be useful to resolve this issue. 

### main branch
When running `python generate_template.py` in the main branch, I get the following error
```
pixel size: 0.04
Traceback (most recent call last):
  File "generate_template.py", line 86, in <module>
    instrument = Instrument('some instrument',
TypeError: __init__() got multiple values for argument 'pixel_size'
```

### JOSS branch
When running `python generate_template.py` in the main branch, I get the following error
```
pixel size: 0.04
Traceback (most recent call last):
  File "generate_template.py", line 47, in <module>
    ext_shear = ExternalShear('my lovely external shear', lens_1.redshift,
NameError: name 'ExternalShear' is not defined
```

It seems the main branch is significantly behind the JOSS branch. I will take this into account for my review and focus on the JOSS branch. If the authors have a different, more up-to-date version of the code, please share it with me.

**Comment by aymgal** [comment]
These example scripts are actually deprecated... sorry for that @AlexandreAdam. Therefore I removed the example directory so that all example notebooks (and related template files) are grouped into a [single place](https://github.com/aymgal/COOLEST/tree/joss/docs/notebooks). I grouped everything under the documentation directory so there are rendered directly in the documentation website.
Let me know if you face problems running the example notebooks.
